pkg/databind/internal/secrets/command.go

```diff
@@ -0,0 +1,89 @@
+// Copyright 2020 New Relic Corporation. All rights reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+package secrets
+
+import (
+	"encoding/json"
+	"errors"
+	"fmt"
+	"os/exec"
+	"strings"
+)
+
+type Command struct {
+	CmdPath string   `yaml:"path"`
+	CmdArgs []string `yaml:"args,omitempty"`
+}
+
+type commandGatherer struct {
+	cfg *Command
+}
+
+var ErrNoPath = errors.New("command secrets must have a path parameter in order to be set")
+
+func validationError(err error) error {
+	return fmt.Errorf("validation error: %w", err)
+}
+
+func runCommandError(err error) error {
+	return fmt.Errorf("failed to run command: %w", err)
+}
+
+func (c *Command) Validate() error {
+	if c.CmdPath == "" {
+		return validationError(ErrNoPath)
+	}
+
+	return nil
+}
+
+// CommandGatherer instantiates a Command variable gatherer from the given configuration. The fetching process
+// will return either a map containing access paths to the stored JSON or a byte string value
+// E.g. if the stored Secret is `{"account":{"user":"test1","password":"test2"}}`, the returned Map
+// contents will be:
+// "account.user"     -> "test1"
+// "account.password" -> "test2".
+func CommandGatherer(cmd *Command) func() (any, error) {
+	cfg := commandGatherer{cmd}
+
+	return func() (any, error) {
+		dt, err := cfg.get()
+		if err != nil {
+			return nil, err
+		}
+
+		return dt, err
+	}
+}
+
+func (c *commandGatherer) get() (any, error) {
+	res, err := runCommand(c.cfg.CmdPath, c.cfg.CmdArgs)
+	if err != nil {
+		return nil, err
+	}
+
+	result := map[string]any{}
+	if err := json.Unmarshal(res, &result); err != nil {
+		slog.WithError(err).Debugf("failed converting command output to json: %v. Returning string", err)
+
+		return strings.TrimSpace(string(res)), nil
```

Comment:
good point here, as we can accept secrets in plain text not in the json format we have specified.
We should make sure this is documented.

I have some doubts about the TrimSpace here. I know is unlikely but what if it has a ' ' instead of '\n'. On the other hand I agree that it is prone to errors since mosts commands will probably return a /n

wdyt if we add a config option in the `command` block for this?

Reply:
also regarding the output type, when is not a json, do we want to return []byte or string?

Reply:
I have changed this, the output for function `runCommand` only deals with `[]byte` output.